homeassistant/components/google_domains/__init__.py

```diff
@@ -42,34 +45,63 @@ async def async_setup(hass, config):
 
     session = hass.helpers.aiohttp_client.async_get_clientsession()
 
+    address = await _get_ip_address(session, timeout)
+    if address is None:
+        return False
+
     result = await _update_google_domains(
-        hass, session, domain, user, password, timeout
+        session, domain, user, password, address, timeout
     )
-
     if not result:
         return False
 
     async def update_domain_interval(now):
         """Update the Google Domains entry."""
-        await _update_google_domains(hass, session, domain, user, password, timeout)
+        nonlocal address
+        new_address = await _get_ip_address(session, timeout)
+        if new_address is not None and new_address != address:
+            result = await _update_google_domains(
+                session, domain, user, password, new_address, timeout
+            )
+            if result:
+                address = new_address
 
     hass.helpers.event.async_track_time_interval(update_domain_interval, INTERVAL)
 
     return True
 
 
-async def _update_google_domains(hass, session, domain, user, password, timeout):
-    """Update Google Domains."""
-    url = f"https://{user}:{password}@domains.google.com/nic/update"
+async def _get_ip_address(session, timeout):
```

Comment:
Please document why we have this function (Google's own IP check is not reliable, etc.)

Reply:
the linked issue seems to be around the component using IPv6 if available, which although from googles perspective is intended behaviour, is breaking things where home assistant is not accessable throught the firewall on v6 it is "unreliable"
 
Using IPify seems to go the other extreme and only return IPv4 addresses as it is a IPv4 only service. ideally the component should get both addresses and write both a A and AAAA. and let happy eyeballs in the browser deal with both being present